framework/src/base/Assembly.C

```diff
@@ -3384,10 +3385,11 @@ Assembly::cacheResidualNodes(const DenseVector<Number> & res,
   // respectively.
   // This is used by NodalConstraint.C to cache the residual calculated for primary and secondary
   // node.
+  const VectorTag & vector_tag = _subproblem.getVectorTag(tag);
```

Comment:
We do not have to change this because it seems no code is passing `tag` in except using its default value 0.

Reply:
I do not want to make that assumption. More general better for me